lib/minitest.rb

```diff
@@ -147,16 +154,26 @@ def self.run args = []
   # sub-classes to run.
 
   def self.__run reporter, options
-    suites = Runnable.runnables.shuffle
-    parallel, serial = suites.partition { |s| s.test_order == :parallel }
-
-    # If we run the parallel tests before the serial tests, the parallel tests
-    # could run in parallel with the serial tests. This would be bad because
-    # the serial tests won't lock around Reporter#record. Run the serial tests
-    # first, so that after they complete, the parallel tests will lock when
-    # recording results.
-    serial.map { |suite| suite.run reporter, options } +
-      parallel.map { |suite| suite.run reporter, options }
+    suites = case suite_order
+    when :random then
+      Runnable.runnables.shuffle
+    else
+      Runnable.runnables.sort_by { |r| r.name }
+    end
```

Comment:
I don't think sorting by name is the correct default because it will reorder people's tests.  Tests are added to `runnables` in load order, which isn't necessarily alphabetic (and if you're using dir globs then it will even change per platform).  If I configure minitest not to muck about with ordering, then I'd expect it to run the tests in the order I specified (especially if I wrote a script to require the tests in a particular order).